packages/js/src/ui/context/NovuContext.tsx

```diff
@@ -0,0 +1,24 @@
+import { createContext, JSX, useContext } from 'solid-js';
+import { Novu, NovuOptions } from '../../novu';
+
+type NovuProviderProps = {
+  options: NovuOptions;
+  children: JSX.Element;
+};
+
+const NovuContext = createContext<Novu | undefined>(undefined);
+
+export function NovuProvider(props: NovuProviderProps) {
+  const novu = new Novu(props.options);
```

Comment:
☑ **chore:** The `novu` object should be memoized we don't want to create a new instance on re-renders